client/src/pages/Scheduler/Scheduler.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState } from 'react';
+import Grid from '@material-ui/core/Grid';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import { Box } from '@material-ui/core';
+import useStyles from './useStyles';
+import { Typography } from '@material-ui/core';
+import { Button } from '@material-ui/core';
+import { useHistory } from 'react-router-dom';
+import Calendar from 'react-calendar';
+import moment from 'moment-timezone';
+import ScheduleIcon from '@material-ui/icons/Schedule';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+
+interface disableDateProp {
+  activeStartDate: Date;
+  date: Date;
+  view: string;
+}
+
+interface TimeZone {
+  [key: string]: { tZone: string; abbr: string };
+}
+
+export default function Scheduler(): JSX.Element {
+  const history = useHistory();
+  const classes = useStyles();
+
+  const username = 'TEMP USER';
+  const duration = 60;
+  const availableDays = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday'];
+  const userTimeZone = 'America/Toronto';
+  const startTime = '08:20';
+  const endTime = '20:00';
+
+  const today = new Date();
+  const [timeZone, setTimeZone] = useState(userTimeZone);
+  const [dateSelected, setDateSelected] = useState(today);
+  const [dateSelISO, setDateSelISO] = useState(today.toISOString());
+
+  // This prop is used to set custom menu for the Select pop-up.
+  const MenuProps = {
+    PaperProps: {
+      style: {
+        maxHeight: 400,
+        width: 250,
+      },
+    },
+  };
+
+  // Create a map of time zones.
+  const timeZones: TimeZone = {};
+  moment.tz.names().map((tZone): void => {
+    const abbr = moment.tz(tZone).zoneName();
+
+    if (abbr[0] === '+' || abbr[0] === '-') return;
+
+    timeZones[tZone] = { tZone, abbr };
+  });
+
+  // Render our list of time zones to choose from.
+  const renderTimeZones = Object.keys(timeZones)
+    .sort()
+    .map((key) => {
+      const { tZone, abbr } = timeZones[key];
+      const time = moment.tz(tZone).format('(h:mma)');
+
+      return (
+        <MenuItem key={tZone} value={tZone}>
+          <h4 style={{ display: 'inline', marginRight: '5px' }}> {`${tZone} (${abbr})`} </h4>
```

Comment:
Code looks fine, great work. Scheduler is a large file, can you refactor the code to implement more modularity like importing _timeButtonBuilder_, And try to remove inline styles, its not a great practice

Reply:
And just to expand on a bit more as to why it's not great practice because we want to stay consistent with styling choices. Inline styling takes priority but since we have the ability to use useStyles for example and it's used widely throughout the app, we'd like to stick to that :)

Reply:
I have refactored the code into three files in total. I also removed inline styling and used a class from useStyles().